### ISSUE TITLE
Allow for Boolean()

### ISSUE THREAD

**Issue by 4nduril-ms** [gkiely/eslint-plugin-jsx-no-leaked-render#3]
Hi, I was just wondering if it would be possible to allow for `Boolean` calls on the left side of  `&&`. I am not that much into ASTs and node types but I understand this would be a CallExpression. Can we somehow "see" what the resulting type of a call expression would be and mark it as a valid condition?

**Comment by gkiely** [comment]
@4nduril-ms calling `Boolean(variable)` does not show an error. E.g.

```js
let x = 0;
const Component = () => {
  return <>{Boolean(x) && <div />}</>;
};
```

Can you provide a code sample that does?

**Comment by 4nduril-ms** [comment]
That's weird.

![Screenshot_20231214_121841](https://github.com/gkiely/eslint-plugin-jsx-no-leaked-values/assets/54028860/bc99ebeb-9cba-44bc-b3c8-0d680103a2ff)

I will check if I can find any settings that are unusual. The project is rather complex (monorepo). 
But  also, when I check out your repo and add the following test case to "valid" it fails:
```
    {
      code: `
      const Component = () => {
        return <>{Boolean(true) && <></>}</>;
      };
    `,
    },
```
